internal/matrix/database/matrix_queries.go

```diff
@@ -1,28 +1,75 @@
 package database
 
+import "fmt"
+
 const (
-	createMatrix    = "create matrix"
-	deleteMatrix    = "delete matrix by uuid"
-	getMatrix       = "get matrix by uuid"
-	getCourseMatrix = "get course and matrix by uuid"
-	listMatrix      = "list matrices"
-	updateMatrix    = "update matrix by uuid"
-	addSubject      = "adds subject to matrix"
-	removeSubject   = "remove subject from matrix"
+	returningMatrixColumns = `matrices.uuid, matrices.code, matrices.name, matrices.description,
+		matrices.created_at, matrices.updated_at`
+	// CREATE.
+	createMatrix = "create matrix"
+	addSubject   = "adds subject to matrix"
+	// READ.
+	getMatrix          = "get matrix by uuid"
+	getCourseMatrix    = "get course and matrix by course_uuid"
+	listMatrix         = "list matrices"
+	listCourseMatrices = "list course matrices by course_uuid"
+	// UPDATE.
+	updateMatrix = "update matrix by uuid"
+	// DELERE.
+	deleteMatrix  = "delete matrix by uuid"
+	removeSubject = "remove subject from matrix"
 )
 
 func queriesMatrix() map[string]string {
 	return map[string]string{
-		createMatrix: "INSERT INTO matrices (name, description) VALUES ($1, $2) RETURNING *",
-		deleteMatrix: "UPDATE matrices SET deleted_at = NOW() WHERE uuid = $1",
-		getMatrix:    "SELECT * FROM matrices WHERE uuid = $1",
-		getCourseMatrix: `SELECT *
+		// CREATE.
+		createMatrix: fmt.Sprintf(`INSERT INTO matrices (
+				course_id, code, name, description
+			) SELECT courses.id, $2, $3, $4
+				FROM courses
+			WHERE courses.uuid = $1 AND courses.deleted_at IS NULL
+			RETURNING %s`, returningMatrixColumns),
+		addSubject: "INSERT INTO matrix_subjects (matrix_id, subject_id) VALUES($1, $2)",
+		// READ.
+		getMatrix: fmt.Sprintf(`SELECT
```

Comment:
I believe that getMatrix and getCourseMatrix could be a single query/method, considering that in the listings queries are returning course information.

Reply:
I updated getCourseMatrix to return if the relationship between course_uuid and matrix_uuid exists